### PR TITLE
feat: 5526 Add newsletter banner in homepage

### PR DESCRIPTION
**Pull request by tihuan**
## Reason for Change

- #5526 
- If the reason for this PR's code changes are not clear in the issue, state value/impact

## Changes

1. This turned out to be a bigger change, because the homepage is the first time we need to support loading two BottomBanner on the same page
2. Add ` isHubSpotReady` and ` onHubSpotReady` on `BottomBanner` to optionally coordinate when HubSpot script is loaded, since `<Script onReady />` only notifies the first `<BottomBanner />` instance with the callback
3. Add `id` to distinguish instances on the same page
4. Use ` formInstanceId` on ` hbspt.forms.create` for HubSpot object to know which instance we're targeting
5. Add `formContainerQueryId` to target different banner instances
6. 

## Testing steps

- Either list QA steps or reasoning you feel QA is unnecessary
- Describe how you made sure to know that your changes worked. Should allow someone else to go verify your code without in depth knowledge.
- "Unit tested only", "tested in rdev by a, b, c, verifying feature worked by... ", "manually ran pipeline locally with these results: ..."

### CODE REVIEW

**Comment by tihuan** [comment]
actually found a bug - will fix tomorrow!

**Comment by tihuan** [comment]
bug fixed!

The bug was that the observer wasn't connecting to the form, because I accidentally set isHubSpotReady back to `false` 🤦

File: frontend/src/components/BottomBanner/index.tsx [hunk 1]
```diff
@@ -75,10 +111,13 @@ export default function BottomBanner({
...
 
-  // eslint-disable-next-line sonarjs/cognitive-complexity
+  /**
+   * Reads a query parameter from the URL to auto open the newsletter signup modal
+   * Allows sharing of a URL to lead directly to the newsletter signup, specifically for conferences
+   */
   useEffect(() => {
-    // Reads a query parameter from the URL to auto open the newsletter signup modal
-    // Allows sharing of a URL to lead directly to the newsletter signup, specifically for conferences
+    if (!isHubSpotReady) return;
```
Comment (tihuan) [inline frontend/src/components/BottomBanner/index.tsx:119]: guard clause for readability - a long block nested in a condition caught me off guard

**Review by tihuan — COMMENTED** [review]
Ready for review PTAL thank you!

**Review by seve — APPROVED** [review]

**Review by kaloster — APPROVED** [review]

**Pull request merged**